### ISSUE TITLE
Can't load Leaderboard

### ISSUE THREAD

**Issue by freeqaz** [facebook/fbctf#351]
Howdy y'all,

Last night, I finishing up some testing for a CTF and ran into some issues getting the Leaderboard working. It looks like there are some issues. Here's a screenshot.

![image](https://i.imgur.com/ucaAiYo.png)

A few of the stack traces (there were many):

```
[Thu Nov 10 12:05:12 2016] [hphp] [19:7eff6afff700:1:000001] [] \nCatchable fatal error: Value returned from async method MultiTeam::genAllTeamsCache() must be of type HH\\Map, bool given in /var/www/fbctf/src/models/MultiTeam.php on line 44
[Thu Nov 10 12:05:12 2016] [hphp] [19:7eff707ff700:7:000001] [] \nCatchable fatal error: Value returned from async method MultiTeam::genLeaderboard() must be of type array, bool given in /var/www/fbctf/src/models/MultiTeam.php on line 73
```

I'm not sure what the issue is because I don't write PHP very much. Hopefully this is an easy one to resolve.

Thanks!

-Free

**Comment by gsingh93** [comment]
@justinwray This seems to be related to the MultiTeam class you added. Can you take a look?

**Comment by justinwray** [comment]
@gsingh93 Roger that.  Will take a look now!

**Comment by justinwray** [comment]
@freeqaz Did you happen to delete all of the teams before this issue arose?  Or did you make any changes to the teams directly in the database?

**Comment by justinwray** [comment]
@gsingh93 This turned out to be a race condition, and is resolved in PR #352.

@freeqaz Thank you for this report!  This should be resolved in the pending PR referenced above.

**Comment by freeqaz** [comment]
Sweet, thanks @justinwray!
